webapp/src/main/java/com/box/l10n/mojito/service/tm/textunitdtocache/TextUnitDTOsCacheService.java

```diff
@@ -211,6 +213,17 @@ Stream<Long> getTmTextUnitIdsForNewTranslations(List<TMTextUnitCurrentVariantDTO
                 .map(TMTextUnitCurrentVariantDTO::getTmTextUnitId);
     }
 
+    Stream<Long> getTmTextUnitIdsForChangedTranslateStatus(ImmutableSet<Long> idsOfDoNotTranslateTextUnits, ImmutableList<TextUnitDTO> textUnitDTOsToUpdate) {
+        return textUnitDTOsToUpdate.stream()
+                .filter(t -> {
+                    boolean doNotTranslateCurrent = idsOfDoNotTranslateTextUnits.contains(t.getTmTextUnitId());
+                    boolean doNotTranslateInNew = t.isDoNotTranslate();
+
+                    return (doNotTranslateCurrent && !doNotTranslateInNew || (!doNotTranslateCurrent && doNotTranslateInNew));
```

Comment:
can it be simplified in

```
boolean newDoNotTranslate = idsOfDoNotTranslateTextUnits.contains(t.getTmTextUnitId());
return newDoNotTranslate != t.isDoNotTranslate();
```

Reply:
Yeah I was thinking whether I should simplify it that way, but I think it makes it not easy to read. What do you think?

Reply:
the simpler the better also to have a consistent use of "new" with the other function. "new" meaning the state from the DB (vs the state from the blob t.xyz())

Reply:
@aurambaj shouldn't it be current actually (as opposed to new) as the DB is only be updated after we detect the change?

Reply:
oh true, okay updated it!